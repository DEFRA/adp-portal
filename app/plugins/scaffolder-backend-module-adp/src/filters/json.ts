import type { JsonValue } from '@backstage/types';

export function json(value: JsonValue) {
  return JSON.stringify(value);
}
