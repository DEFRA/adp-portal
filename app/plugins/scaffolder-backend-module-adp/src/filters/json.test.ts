import type { JsonValue } from '@backstage/types';
import { json } from './json';

const inputs: JsonValue[] = [
  123,
  'abc',
  true,
  false,
  null,
  `This is some multiline text
and this is line 2!`,
  `This input: contains special characters! "|'{}[],&*#?<>=%@/\\
And even a newline!`,
  '"This already looks like json but should be escaped"',
  {
    some: {
      object: {
        with: ['values'],
      },
    },
  },
];

interface TestCase {
  readonly input: JsonValue;
  readonly inputJSON: string;
}

describe('json', () => {
  it.each<TestCase>(
    inputs.map(i => ({ input: i, inputJSON: JSON.stringify(i) })),
  )('Should correctly handle $inputJSON', ({ input }) => {
    // act
    const actual = json(input);

    // assert
    expect(actual).toMatchSnapshot();
  });
});
