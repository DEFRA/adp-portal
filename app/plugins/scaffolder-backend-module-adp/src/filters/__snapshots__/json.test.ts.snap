// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json Should correctly handle "\\"This already looks like json but should be escaped\\"" 1`] = `""\\"This already looks like json but should be escaped\\"""`;

exports[`json Should correctly handle "This input: contains special characters! \\"|'{}[],&*#?<>=%@/\\\\\\nAnd even a newline!" 1`] = `""This input: contains special characters! \\"|'{}[],&*#?<>=%@/\\\\\\nAnd even a newline!""`;

exports[`json Should correctly handle "This is some multiline text\\nand this is line 2!" 1`] = `""This is some multiline text\\nand this is line 2!""`;

exports[`json Should correctly handle "abc" 1`] = `""abc""`;

exports[`json Should correctly handle {"some":{"object":{"with":["values"]}}} 1`] = `"{"some":{"object":{"with":["values"]}}}"`;

exports[`json Should correctly handle 123 1`] = `"123"`;

exports[`json Should correctly handle false 1`] = `"false"`;

exports[`json Should correctly handle null 1`] = `"null"`;

exports[`json Should correctly handle true 1`] = `"true"`;
